proof_gen/src/proof_gen.rs

```diff
@@ -70,17 +70,23 @@ pub fn generate_segment_proof(
 /// Generates an aggregation proof from two child proofs.
 ///
 /// Note that the child proofs may be either transaction or aggregation proofs.
+///
+/// If a transaction only contains a single segment, this function must still be
+/// called to generate a `GeneratedSegmentAggProof`. In that case, you can set
+/// `has_dummy` to `true`, and provide an arbitrary proof for the right child.
```

Comment:
For the `arbitrary` proof, `proof_gen` is re-implementing the `dummy_circuit` / `dummy_proof` methods of plonky2 that are now being exposed, so could be nice to reuse them directly instead with a re-export.

Reply:
I'm sorry, I'm not sure I understand what this refers to/ what you mean here

Reply:
`proof_gen` reimplements `dummy_proof` to be used in `test_only` mode of `zero-bin`. We could just do a re-export of the now public `plonky2` method so that we don't need to implement twice the same method.

Reply:
Since the Plonky2 `dummy_proof` requires two arguments, I changed the implementation in `proof_gen` to reuse the Plonky2 function (building a dummy circuit before).